marketpulse/main/views.py

```diff
@@ -168,3 +168,56 @@ def manifest(request):
                             content_type='application/x-web-app-manifest+json; charset=utf-8')
     response['Content-Disposition'] = 'attachment;filename=manifest.webapp'
     return response
+
+
+@login_required
+def edit_fxosmedia(request):
+    user = request.user
+
+    if request.is_ajax():
```

Comment:
I suggest instead of copying code from other views to add a helper method and use it both here and in the other activity type.